test/test_port.cpp

```diff
@@ -0,0 +1,63 @@
+/*********************************************************************
+ *        _       _         _
+ *  _ __ | |_  _ | |  __ _ | |__   ___
+ * | '__|| __|(_)| | / _` || '_ \ / __|
+ * | |   | |_  _ | || (_| || |_) |\__ \
+ * |_|    \__|(_)|_| \__,_||_.__/ |___/
+ *
+ * www.rt-labs.com
+ * Copyright 2020 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include "utils_for_testing.h"
+#include "mocks.h"
+
+#include "pf_includes.h"
+
+#include <gtest/gtest.h>
+
+class PortUnitTest : public PnetUnitTest
+{
+};
+
+TEST_F (PortUnitTest, PortCheckItetator)
+{
+   pnet_t netstruct; /* Do not care about actual value */
+   pf_port_iterator_t port_iterator;
+   int port;
+   int ix;
+
+   /* Check precondiiton. Ports must be enabled for the stack to be working. */
+   EXPECT_TRUE (PNET_MAX_PORT > 0);
+
+   /* Retrieve first port */
+   pf_port_init_port_iterator (&netstruct, &port_iterator);
+   port = pf_port_get_next_port (&port_iterator);
+   EXPECT_EQ (port, 1);
+
+   /* More ports might be available dependent on compile time setting */
+   for(ix = 2; ix <= PNET_MAX_PORT; ix++)
+   {
+      port = pf_port_get_next_port (&port_iterator);
+      EXPECT_EQ (port, ix);
+   }
+
+   /* Verify that we return 0 when we are done */
+   port = pf_port_get_next_port (&port_iterator);
+   EXPECT_EQ (port, 0);
+   port = pf_port_get_next_port (&port_iterator);
+   EXPECT_EQ (port, 0);
+   port = pf_port_get_next_port (&port_iterator);
+   EXPECT_EQ (port, 0);
+   port = pf_port_get_next_port (&port_iterator);
+   EXPECT_EQ (port, 0);
```

Comment:
Remove these as testing the same thing four times is a bit excessive.

Reply:
Done